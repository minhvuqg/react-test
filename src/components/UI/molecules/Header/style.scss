@import 'scss/var/color.scss';

.header {
  width: 100vw;
  background-color: $primary-color;

  > .-container {
    display: flex;
    align-items: center;
    height: 48px;
    justify-content: space-between;
    padding: 0 32px;

    .logo {
      width: 32px;

      img {
        width: 100%;
        height: auto;
      }
    }

    .-text-link {
      color: #fff;
      position: relative;
      margin: auto;
      padding: 3px 5px;
      &:after {
        content: '';
        position: absolute;
        width: 100%;
        transform: scaleX(0);
        height: 2px;
        bottom: 0;
        left: 0;
        background-color: #fff;
        transform-origin: bottom right;
        transition: transform 0.4s ease-in-out;
      }
      &:hover {
        cursor: pointer;
        &:after {
          transform: scaleX(1);
          transform-origin: bottom left;
        }
      }
    }
  }
}
