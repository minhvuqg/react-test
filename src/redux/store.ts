import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
