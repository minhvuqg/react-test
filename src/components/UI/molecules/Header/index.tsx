import { FC } from 'react'
import { Link } from 'react-router-dom'

import { IProps } from './type'

import './style.scss'

export const Header: FC<IProps> = (): JSX.Element => {
  return (
    <header className="header">
      <div className="-container">
        <div className="logo">
          <Link to="/" data-testid="logo">
            <img src="/imgs/logo.png" alt="Logo" />
          </Link>
        </div>
        <div className="link-list">
          <Link data-testid="home" to="/" className="p-r-16 -text-link">
            Home
          </Link>
          <Link data-testid="cart" to="/cart" className="-text-link">
            Cart
          </Link>
        </div>
      </div>
    </header>
  )
}

export default Header
