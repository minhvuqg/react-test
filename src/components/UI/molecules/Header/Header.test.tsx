import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Header from '../Header'
import store from 'redux/store'
import { BrowserRouter as Router } from 'react-router-dom'

it('render header correctly', () => {
  const headerEl = TestRenderer.create(
    < Provider store={store} >
      <Router>
        <Header />
      </Router>
    </Provider >).toJSON()
  expect(headerEl).toMatchSnapshot()
})
