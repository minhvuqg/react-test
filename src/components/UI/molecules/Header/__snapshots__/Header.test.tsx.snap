// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render header correctly 1`] = `
<header
  className="header"
>
  <div
    className="-container"
  >
    <div
      className="logo"
    >
      <a
        data-testid="logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          src="/imgs/logo.png"
        />
      </a>
    </div>
    <div
      className="link-list"
    >
      <a
        className="p-r-16 -text-link"
        data-testid="home"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="-text-link"
        data-testid="cart"
        href="/cart"
        onClick={[Function]}
      >
        Cart
      </a>
    </div>
  </div>
</header>
`;
